tempesta_fw/http.c

```diff
@@ -569,15 +576,19 @@ tfw_http_req_cache_cb(TfwHttpReq *req, TfwHttpResp *resp)
 
 	/* Send request to the server. */
 	tfw_connection_send(conn, (TfwMsg *)req, false);
-
 	goto conn_put;
+
 send_404:
 	tfw_http_send_404((TfwHttpMsg *)req);
 	tfw_http_conn_msg_free((TfwHttpMsg *)req);
+	if (req_flags & TFW_HTTP_CONN_CLOSE)
+		ss_droplink(req_sk);
 	return;
 send_500:
 	tfw_http_send_500((TfwHttpMsg *)req);
 	tfw_http_conn_msg_free((TfwHttpMsg *)req);
+	if (req_flags & TFW_HTTP_CONN_CLOSE)
+		ss_droplink(req_sk);
```

Comment:
Why not to call

	if (req_flags & TFW_HTTP_CONN_CLOSE)
		ss_droplink(req_sk);

from ```tfw_http_conn_msg_free()``` or some other wrapping function? If we drop the connection on SS layer, then we also should drop TfwConnection and associated messages.